core/field_label_editable.js

```diff
@@ -61,3 +61,38 @@ Blockly.FieldLabelEditable.prototype.updateWidth = function() {
   // Unlike the base Field class, this doesn't add space to editable fields.
   this.size_.width = Blockly.Field.getCachedWidth(this.textElement_);
 };
+
+/**
+ * Draws the border with the correct width.
+ * Saves the computed width in a property.
+ * @private
+ */
+Blockly.FieldLabelEditable.prototype.render_ = function() {
```

Comment:
So this is very similar to the render function of the base `Field` right? Why doesn't the base class rendering function center the label correctly? Could you explain a bit more about why this was needed?

Reply:
The base class has the following code around the centering:
```

    // In a text-editing shadow block's field,
    // if half the text length is not at least center of
    // visible field (FIELD_WIDTH), center it there instead,
    // unless there is a drop-down arrow.
    if (this.sourceBlock_.isShadow() && !this.positionArrow) {
```

So it won't center the label correctly if it's not a shadow block.  The check in the base class seems really suspect to me anyway--it claims to be for a "text-editing shadow block's field", and `Field` is probably the wrong place for that code.  It could probably go somewhere much more specific.

Rather than add more checks in Field, I just overwrote this to get the centering right for this particular field.  I removed some code that I don't need--mostly around the arrow and the box.

Reply:
Ok sounds good.